sdk/core/System.ClientModel/src/Options/ClientLoggingOptions.cs

```diff
@@ -0,0 +1,283 @@
+                                                                          "Pragma",
+                                                                          "Request-Id",
+                                                                          "Retry-After",
+                                                                          "Server",
+                                                                          "Transfer-Encoding",
+                                                                          "User-Agent",
+                                                                          "WWW-Authenticate" ];
+    private static readonly HashSet<string> s_defaultAllowedQueryParameters = ["api-version"];
+    private static readonly PipelineMessageSanitizer s_defaultSanitizer = new(s_defaultAllowedQueryParameters, s_defaultAllowedHeaderNames);
+
+    internal const bool DefaultEnableLogging = true;
+    internal const bool DefaultEnableMessageContentLogging = false;
+    internal const int DefaultMessageContentSizeLimitBytes = 4 * 1024;
+    internal const double RequestTooLongSeconds = 3.0; // sec
+
+    /// <summary>
+    /// Gets or sets the implementation of <see cref="ILoggerFactory"/> to use to
+    /// create <see cref="ILogger"/> instances for logging.
+    /// </summary>
+    /// <remarks>If an ILoggerFactory is not provided, logs will be written to Event Source
+    /// instead. If an ILoggerFactory is provided, logs will be written to ILogger only and not
+    /// Event Source.</remarks>
+    /// <value>Defaults to <see cref="NullLoggerFactory"/>.</value>
+    public ILoggerFactory? LoggerFactory
+    {
+        get => _loggerFactory;
+        set
+        {
+            AssertNotFrozen();
+
+            _loggerFactory = value;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets value indicating if logging should be enabled in this client pipeline.
+    /// </summary>
+    /// <value>Defaults to <c>null</c>. If <c>null</c>, this value will be treated as <c>true</c>.</value>
+    public bool? EnableLogging
+    {
+        get => _enableLogging;
+        set
+        {
+            AssertNotFrozen();
+
+            _enableLogging = value;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets value indicating if request and response content should be logged.
+    /// </summary>
+    /// <value>Defaults to <c>null</c>. If <c>null</c>, the value
+    /// of <see cref="EnableLogging"/> will be used instead.</value>
+    public bool? EnableMessageLogging
+    {
+        get => _enableMessageLogging;
+        set
+        {
+            AssertNotFrozen();
+
+            _enableMessageLogging = value;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets value indicating if request and response content should be logged.
+    /// </summary>
+    /// <value>Defaults to <c>null</c>. If <c>null</c>, this value will be treated as <c>false</c>.</value>
+    public bool? EnableMessageContentLogging
```

Comment:
What is the difference between this property and EnableMessageLogging? The descriptions look the same.

Reply:
There's an error for the EnableMessageLogging doc that I need to fix, thanks for pointing it out. This one is for request/response content and EnableMessageLogging is for general request and response logging not including content.